lib/reporters/default.js

```diff
@@ -0,0 +1,252 @@
+'use strict';
+const os = require('os');
+const path = require('path');
+const stream = require('stream');
+
+const cliCursor = require('cli-cursor');
+const figures = require('figures');
+const indentString = require('indent-string');
+const ora = require('ora');
+const plur = require('plur');
+const trimOffNewlines = require('trim-off-newlines');
+const trimRight = require('trim-right');
+
+const chalk = require('../chalk').get();
+const codeExcerpt = require('../code-excerpt');
+const colors = require('./colors');
+const formatSerializedError = require('./format-serialized-error');
+const improperUsageMessages = require('./improper-usage-messages');
+const prefixTitle = require('./prefix-title');
+const whileCorked = require('./while-corked');
+
+class LineWriter extends stream.Writable {
+	constructor(dest, spinner) {
+		super();
+
+		this.dest = dest;
+		this.columns = dest.columns || 80;
+		this.spinner = spinner;
+		this.lastSpinnerText = '';
+	}
+
+	_write(chunk, encoding, callback) {
+		// Discard the current spinner output. Any lines that were meant to be
+		// preserved should be rewritten.
+		if (this.spinner) {
+			this.spinner.clear();
+		}
+
+		this._writeWithSpinner(chunk);
+		callback();
+	}
+
+	_writev(pieces, callback) {
+		// Discard the current spinner output. Any lines that were meant to be
+		// preserved should be rewritten.
+		if (this.spinner) {
+			this.spinner.clear();
+		}
+
+		const last = pieces.pop();
+		for (const piece of pieces) {
+			this.dest.write(piece.chunk);
+		}
+		this._writeWithSpinner(last.chunk);
+		callback();
+	}
+
+	_writeWithSpinner(chunk) {
+		if (!this.spinner || !this.spinner.id) {
```

Comment:
With latest Ora, you can check `this.spinner.isSpinning` instead of `this.spinner.id`: https://github.com/sindresorhus/ora/commit/3df8d0d962b3af9aa5c269ed9ac3d21734b34d32